### PR TITLE
Add build_info metric and use it in generated queries 

### PR DESCRIPTION
**Pull request by brettimus**
Implements #25

***

## Description for Reviewers

**We should discuss how we want to pull in the commit and version info, especially in relation to the expected behavior in VSCode.**

### What Changed

- Add an instance method of `set_build_info` to both the Prometheus and Otel trackers
  - Make sure that `set_build_info` is only called once (per tracker instance)
  - Note that the OTEL tracker implements build_info as an `UpDownCounter`.

- When `create_tracker` is called, we will automatically call `set_build_info` with the `AUTOMETRICS_COMMIT` and `AUTOMETRICS_VERSION` environment variables

- Update PromQL queries to use commit and version info

## TODO
- [ ] Update README regarding the environment variables we want to support for version/commit
- [x] Check whether OTEL tracker should also use the name `build_info`
- [x] Only call tracker `set_build_info` once

### CODE REVIEW

**Comment by brettimus** [comment]
Update: OTEL support will be dependent on resolution of https://github.com/open-telemetry/opentelemetry-python/issues/3071 (I'm working on a PR)

**Comment by brettimus** [comment]
I think we can treat the OTEL support as non-blocking, since I'm not sure how long it will take for my PR to get reviewed/merged/released

https://github.com/open-telemetry/opentelemetry-python/pull/3306

**Review by flenter — COMMENTED** [review]
🎉 Nice! 

However, I think it would be good to expand the `TrackerType` protocol (i.e. interaface) with a `set_build_info` function (defined in `src/autometrics/tracker/tracker.py`)

File: src/autometrics/test_prometheus_url.py [hunk 1]
```diff
@@ -25,10 +25,12 @@ def test_create_urls_with_default_url(default_url_generator: Generator):
     urls = default_url_generator.create_urls()
 
     # print(urls.keys())
+    # print(urls.values())
```
[the remark this answers is not in the document]
Comment (brettimus) [reply src/autometrics/test_prometheus_url.py]: wham bam it's gone!

File: src/autometrics/prometheus_url.py [hunk 2]
```diff
@@ -3,6 +3,8 @@
 from typing import Optional
 from dotenv import load_dotenv
 
+ADD_BUILD_INFO_LABELS = "* on (instance, job) group_left(version, commit) build_info"
```
[the remark this answers is not in the document]
Comment (brettimus) [reply src/autometrics/prometheus_url.py]: ah good catch! i updated the queries + tests

File: src/autometrics/tracker/tracker.py [hunk 3]
```diff
@@ -37,16 +40,26 @@ class TrackerType(Enum):
...
+        tracker_instance = OpenTelemetryTracker()
     elif tracker_type == TrackerType.PROMETHEUS:
         # pylint: disable=import-outside-toplevel
         from .prometheus import PrometheusTracker
 
-        return PrometheusTracker()
+        tracker_instance = PrometheusTracker()
+
+    # NOTE - Only set the build info when the tracker is initialized
+    if tracker_instance:
```
[the remark this answers is not in the document]
Comment (brettimus) [reply src/autometrics/tracker/tracker.py:56]: ah right i wasn't thinking in types. will fix this, then add a blurb to the README about how otel support is coming for build_info, then will merge!

**Pull request merged**